### ISSUE TITLE
Add "check" build to speed test turn around

### ISSUE THREAD

**Issue by russellhadley** [dotnet/llilc#511]
Add a 'release with asserts' check build speed up testing time for LLILC

**Comment by AndyAyersMS** [comment]
@adiaaida @mmitche @russellhadley are we getting anywhere on implementing this or on upping the CI timeout needed to unblock #496?

I worry because we are doing very minimal testing of LLILC right now. Some of the new tests evidently fail if LLILC is asked to jit everything so once we can run everything in the lab there's still an issue tail to work through.

Fixing this really needs to be our top priority.

**Comment by mmitche** [comment]
@dotnet-bot test this please

I upped the timeout to 240.  Still working on the side to get this thing under the profiler.

**Comment by adiaaida** [comment]
Is the problem that we are timing out for individual tests, or is the CI timing out? These are 2 different issues. If the former, what @mmitche did should clear it up. If the later, then I think I figured out what needs to be updated in coreclr to increase the individual test time out?

**Comment by mmitche** [comment]
It might be both.  I fixed the timeout for the whole CI job.

**Comment by adiaaida** [comment]
It looks like it's just an issue with the timeout for the whole job.

**Comment by adiaaida** [comment]
I've added a new pr test job called dotnet_llilc_check_win64_prtest, which does a RelWithDebInfo build with asserts on.

**Comment by russellhadley** [comment]
@adiaaida can we close this?

**Comment by JosephTremoulet** [comment]
How much faster is the "RelWithDebInfo build with asserts on" configuration?  Are the instructions for building that way up on the wiki?

**Comment by adiaaida** [comment]
@russellhadley I've created the job but haven't added it as a leg/replaced the debug leg with it.
@JosephTremoulet the last time I checked, maybe 10 minutes.

**Comment by adiaaida** [comment]
I've switched us over to the check job for pr testing. When someone submits a PR, I'll make sure it's running correctly.

**Comment by adiaaida** [comment]
I misspoke: the check build takes almost exactly the same amount of time as the debug build.

**Comment by pgavlin** [comment]
I noticed that the job is still configured to pass `/p:configuration=Debug` to MSBuild. Are there any other configurations available when the projects are generated with `-DCMAKE_BUILD_TYPE=RelWithDebInfo`?

**Comment by pgavlin** [comment]
I've verified locally that configuring with `-DCMAKE_ BUILD_TYPE=RelWithDebInfo` produces a .sln with `Debug`, `Release`, `MinSizeRel`, and `RelWithDebInfo` configurations. We should probably switch the configuration over and try again.

**Comment by pgavlin** [comment]
Looks like the check build sped things up tremendously: the test run portion of the build now takes ~10 minutes (see http://dotnet-ci.cloudapp.net/job/dotnet_llilc_check_win64_prtest/8/consoleFull for an example).

**Comment by pgavlin** [comment]
I'm going to close this now. Please feel free to reopen if I am mistaken in doing so.

**Comment by JosephTremoulet** [comment]
What exactly are the steps to use the check build locally?  Seems like they should be recorded on the wiki, though I don't know if it should be as an addition or change to the build steps (which we have separately for [Windows](https://github.com/dotnet/llilc/wiki/Getting-Started-For-Windows), [Linux and OS X](https://github.com/dotnet/llilc/wiki/Getting-Started-For-Linux-and-OS-X), and [Out-of-tree](https://github.com/dotnet/llilc/wiki/Setup-With-LLILC-Out-Of-LLVM-Tree)), or as a note in the [testing page](https://github.com/dotnet/llilc/wiki/Testing).

**Comment by adiaaida** [comment]
To get a check build, you would run cmake with the following arguments: -DCMAKE_BUILD_TYPE=RELWITHDEBINFO -DLLVM_ENABLE_ASSERTIONS=ON

and msbuild with /p:Configuration=RelWithDebInfo

**Comment by JosephTremoulet** [comment]
Ok... I copied that over to the [testing page](https://github.com/dotnet/llilc/wiki/Testing), thanks.
